plugins/flytekit-envd/flytekitplugins/envd/image_builder.py

```diff
@@ -52,14 +57,16 @@ def build():
     runtime.environ(env={env})
     config.pip_index(url = "{pip_index}")
 """
+    ctx = context_manager.FlyteContextManager.current_context()
+    cfg_path = ctx.file_access.get_random_local_path("build.envd")
+    pathlib.Path(cfg_path).parent.mkdir(parents=True, exist_ok=True)
 
     if image_spec.python_version:
         # Indentation is required by envd
         envd_config += f'    install.python(version="{image_spec.python_version}")\n'
 
-    ctx = context_manager.FlyteContextManager.current_context()
-    cfg_path = ctx.file_access.get_random_local_path("build.envd")
-    pathlib.Path(cfg_path).parent.mkdir(parents=True, exist_ok=True)
+    if image_spec.cuda and image_spec.cudnn:
+        envd_config += f'    install.cuda(version="{image_spec.cuda}", cudnn="{image_spec.cudnn}")\n'
```

Comment:
the [docs](https://envd.tensorchord.ai/api/starlark/v0/install.html#cuda) mention that `cudnn` by default this is set to `8`. Should we check the value of `image_spec.cudnn`?

Reply:
envd will help us to check that. If people use the wrong cudnn version, envd will raise an error.

Reply:
Sorry, I just realized cudnn is just an extra plugin. Update it, and make it optional